packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -402,13 +402,15 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
     /**
      * @inheritdoc IAssetManager
      */
-    function undelegate(address validator, uint128 shares) external {
+    function undelegate(address validator, uint128 assets) external {
```

Comment:
Maybe we don't need the function `getKroTotalShareBalance` which was added for external purpose, wdyt?

Reply:
Also I think we don't need `previewDelegate`, `previewUndelegate`. Instead, how about adding `getKroAssets` to fetch total asset of a delegator?

Reply:
The interface should be changed corresponding to this function.

Reply:
Yeah, we don't need `getKroTotalShareBalance`, so we can remove it. 
AFAIK, `previewDelegate` and `previewUndelegate` are interface of ERC-4626, so added. I discussed about these before with @sm-stack, and concluded that follow the interface with implementation.

Reply:
Yeah I know that it is the interface of ERC-4626, but we actually don't extend the interface and `share` is only used internally, I thought it doesn't need to exist. wdyt?

Reply:
Although the interfaces changed in many parts compared to ERC-4626, I think those functions have their roles for checking the status of my current delegations and further delegations. For example, combining `getKroTotalShareBalance` and `previewUndelegate` may be used to query how much assets I delegated. That may help users to check their funds are still there even if the frontend is unavailable in some reasons. How do you think about this?

Reply:
I think it's better to just add `getKroAssets` function which do the things internally what you said (query how much assets I delegated).

Reply:
Why don't we have all those things? `previewUndelegate`, `previewDelegate`, and `getKroAssets`.

Reply:
I'm still not sure if we need `previewDelegate` and `previewUndelegate`, but I think we should keep `getKroTotalShareBalance` also if we keep them.

Reply:
Ok. Let's just leave as is for future compatibility. I'll add `getKroAssets` in this PR.

Reply:
Wait, I think we cannot make `getKroAssets` to get all KRO delegated, since there is no way to get all address of validator at all in the contract. Am I missing something?

Reply:
Right, I thought `getKroAssets(validator, delegator)`

Reply:
Ah, I see. I confused with `totalKroAssets`, so I thought that we don't need new one. I'll add it.

Reply:
Added in [6692470](https://github.com/kroma-network/kroma/pull/350/commits/669247060137c19dc2532db248bb709eac9a676d)